src/NuGet.DependencyResolver.Core/Remote/RemoteDependencyWalker.cs

```diff
@@ -144,7 +139,7 @@ public Task<GraphNode<RemoteResolveResult>> WalkAsync(LibraryRange library, NuGe
 
             if (match == null)
             {
-                return null;
+                return CreateUnresolvedMatch(libraryRange);
```

Comment:
@davidfowl Rather than having a special "Unresolved" dependency provider, it seemed more appropriate to have the Walker handle unresolved dependencies itself, since it's a core concept.